cfme/common/provider.py

```diff
@@ -514,12 +442,22 @@ def get_credentials(cls, credential_dict, cred_type=None):
         """
         domain = credential_dict.get('domain', None)
         token = credential_dict.get('token', None)
-        return cls.Credential(
-            principal=credential_dict['username'],
-            secret=credential_dict['password'],
-            cred_type=cred_type,
-            domain=domain,
-            token=token)
+        if not cred_type:
```

Comment:
Actually, I have a set of consequent changes. This is first step.
I'm planning to get rid of this statement in #4463, one of next steps.